benches/benches.rs

```diff
@@ -30,36 +31,75 @@ async fn ws_server(tx: Sender<SocketAddr>) {
 	}
 }
 
-pub fn http(c: &mut criterion::Criterion) {
-	c.bench_function("http 100 requests", |b| {
-		let (tx_addr, rx_addr) = oneshot::channel::<SocketAddr>();
-		async_std::task::spawn(http_server(tx_addr));
-		let server_addr = block_on(rx_addr).unwrap();
-		let client = HttpClient::new(&format!("http://{}", server_addr));
+pub fn http_requests(c: &mut criterion::Criterion) {
+	let mut rt = tokio::runtime::Runtime::new().unwrap();
+	let (tx_addr, rx_addr) = oneshot::channel::<SocketAddr>();
+	async_std::task::spawn(http_server(tx_addr));
+	let server_addr = block_on(rx_addr).unwrap();
+	let client = Arc::new(HttpClient::new(&format!("http://{}", server_addr)));
 
+	c.bench_function("synchronous http round trip", |b| {
 		b.iter(|| {
-			block_on(async {
-				for _ in 0..100 {
-					let _: JsonValue = black_box(client.request("say_hello", Params::None).await.unwrap());
-				}
+			rt.block_on(async {
+				let _: JsonValue = black_box(client.request("say_hello", Params::None).await.unwrap());
 			})
 		})
 	});
+
+	c.bench_function_over_inputs(
+		"concurrent http round trip",
+		move |b: &mut Bencher, size: &usize| {
+			b.iter(|| {
+				let mut tasks = Vec::new();
+				for _ in 0..*size {
+					let client_rc = client.clone();
+					let task = rt.spawn(async move {
+						let _: Result<JsonValue, _> = black_box(client_rc.request("say_hello", Params::None)).await;
+					});
+					tasks.push(task);
+				}
+				for task in tasks {
+					rt.block_on(task).unwrap();
+				}
+			})
+		},
+		vec![2, 4, 8, 16, 32, 64, 128],
```

Comment:
I don't understand what the point of measuring this over different number of loops is? Is it to see what performance is when number of tasks is higher than the number of threads in the thread pool? Or…?

Reply:
It is a way to measure how an increasing number of concurrent tasks impact performance,  such as try 2 concurrent request, .... to 128 if that makes sense. That's at least the intention

Thus, tasks is requests in this context.

Reply:
Ok, I see what you're saying. Maybe it should check core count and do like `let cores = …; let batches = [cores/4, cores/2, cores, cores * 2, cores * 4]`?
It is also measuring *roundtrips* so I guess it's ok to include the client part too. We should still have benchmarks for the actual execution at some point, but maybe that should be decoupled from the async machinery.